MetabolicAtlas/MetabolicAtlas
# API for 2D SVG map thumbnails

This PR adds the API endpoint for 2D map thumbnails.

Description, an thumbnail in PNG format is returned by accessing the API endpoint for a specified 2D map. 

To test:
    /api/v2/svg-thumbnail/acylglycerides_metabolism?model=Human-GEM

You will need to rebuild the stack in order to test this pull request since a new package `sharp` has been added to the package list. 

The thumbnail image is generated by the package [sharp](https://www.npmjs.com/package/sharp) in real time. 
To be discussed: 
1. Size of the thumbnail image. Currently, all thumbnails are with a fixed width of 400px. (Resolved)
    Width of the thumbnail can be set by the parameter `width` in the query, although it should be between 100 and 600.
2. Format of the return value of the API endpoint. Currently, the return value is a png image. (Resolved)
    PNG is a proper format for thumbnail.
3. In the converted thumbnail png images, characters are all displayed as squares since fonts such as`Arial` are not available in the Docker container. To install these fonts may increase the size of the container significantly, not to say some of the fonts are not free according to https://wiki.alpinelinux.org/wiki/Fonts. For thumbnails, when the width is around 400, letters can not be seen, so probably we can ignore this.
